functionary/core/admin/team.py

```diff
@@ -0,0 +1,69 @@
+from django import forms
+from django.contrib import admin
+
+from core.models import Environment, Team, TeamUserRole
+
+
+class EnvironmentChoiceField(forms.ModelMultipleChoiceField):
+    """Custom choice field to override the name of the Environment"""
+
+    def label_from_instance(self, obj):
+        return obj.name
+
+
+class UserRoleInline(admin.TabularInline):
+    """Inline form to add users to the Team"""
+
+    model = TeamUserRole
+    extra = 0
+    verbose_name = "User"
+
+    def has_change_permission(self, request, obj):
+        return False
+
+    def formfield_for_dbfield(self, db_field, request, **kwargs):
+        field = super().formfield_for_dbfield(db_field, request, **kwargs)
+        # By default, Django wraps the widget to allow you to create these
+        # objects inline with the form. Unwrap the widget since this is not
+        # what we want.
+        if db_field.attname == "user_id":
+            field.widget = field.widget.widget
+        return field
+
+
+class TeamForm(forms.ModelForm):
+    """Custom form to show the environments associated with the team"""
+
+    environments = EnvironmentChoiceField(
```

Comment:
This ends up getting displayed as a multi-select box where your selection doesn't do anything.  Something like the list that you have for users, except it's read-only and just links over to the environments might be more appropriate.

Reply:
Changed the list to readonly. I didn't see a widget that I could change it to in order to just display links. For a field on a model, that's handled by rendering a div on a template. This field is only defined on the form and makes things more challenging.